apps/rig/config/config.exs

```diff
@@ -58,6 +58,9 @@ config :rig, Rig.EventStream.KafkaToHttp,
   targets: {:system, :list, "FIREHOSE_KAFKA_HTTP_TARGETS", []},
   group_id: {:system, "KAFKA_GROUP_ID", "rig"}
 
+config :rig, Rig.Connection.Codec,
+  codec_secret_key: {:system, "NODE_COOKIE", "7tsf4Y6GTOfPY1iDo4PqZA=="}
```

Comment:
hm `mix compile` _should_ tell you that `NODE_COOKIE` cannot have more than one default value. Please use `nil` as the default, too. You can use this (or a random) string in the code below in case `NODE_COOKIE` is not set.

Reply:
try running `MIX_ENV=prod mix compile`, which should give you the following change:

```bash
$ git diff
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
modified: docs/rig-ops-guide.md
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
@ docs/rig-ops-guide.md:60 @ Variable&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
`KINESIS_OTP_JAR` | Path to the `OtpErlang.jar` file that contains the `JInterface` implementation. If left empty, RIG picks the file from its Erlang environment (Erlang must be compiled with Java support enabled). | nil
`KINESIS_STREAM` | The name of the Kinesis stream to consume. | "RIG-outbound"
`LOG_LEVEL` | Controls logging level for RIG, available values are: "debug", "info", "warn", "error". Production is using "warn" level. | :warn
`NODE_COOKIE` | Erlang cookie used in distributed mode, so nodes in cluster can communicate between each other. | nil
`NODE_COOKIE` | Erlang cookie used in distributed mode, so nodes in cluster can communicate between each other. | "7tsf4Y6GTOfPY1iDo4PqZA=="
`NODE_HOST` | Erlang hostname for given node, used to build Erlang long-name `rig@NODE_HOST`. This value is used by Erlang's distributed mode, so nodes can see each other. | nil
`PROXY_CONFIG_FILE` | Configuration JSON file with initial API definition for API Proxy. Use this variable to pass either a path to a JSON file, or the JSON string itself. A path can be given in absolute or in relative form (e.g., `proxy/your_json_file.json`). If given in relative form, the working directory is one of RIG's `priv` dirs (e.g., `/opt/sites/rig/lib/rig_inbound_gateway-2.0.2/priv/` in a Docker container). | nil
`PROXY_RECV_TIMEOUT` | Timeout used when receiving a response for a forwarded/proxied request. | 5000
```

Reply:
just noticed that (1) you need to run mix compile with `MIX_ENV=prod` for the script to do anything and (2) the script ignores the `nil` default and uses the default you define here as the default for both settings \*sigh\*

So please just make sure both are set to `nil` :)

Reply:
@pavestru update on this?

Reply:
I've set the default value here to nil, added another default value in the same config and used it in code as follows:
```elixir
secret_key = conf.codec_secret_key || conf.codec_default_key
```
see [the commit](https://github.com/pavestru/reactive-interaction-gateway/commit/632b9969b5283c8b2f8eb6dcfe9cbe267b213ed7)